kube/secret.go

```diff
@@ -0,0 +1,101 @@
+package kube
+
+import (
+	"fmt"
+	//	"strconv"
+	"strings"
+
+	"github.com/SUSE/fissile/model"
+	meta "k8s.io/client-go/pkg/api/unversioned"
+	apiv1 "k8s.io/client-go/pkg/api/v1"
+)
+
+// NewSecret creates a single new, empty K8s Secret
+func NewSecret(name string) *apiv1.Secret {
+	secret := &apiv1.Secret{
+		TypeMeta: meta.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Secret",
+		},
+		ObjectMeta: apiv1.ObjectMeta{
+			Name: name,
+		},
+	}
+	secret.Data = make(map[string][]byte)
+	return secret
+}
+
+// Ref is an entry in the RefMap
+type Ref struct {
+	Secret string
+	Key    string
+}
+
+// RefMap maps the names of secret CVs to the combination of secret
+// and key used to store their value. Note that the key has to be
+// stored, because of the transformation at (**). Ok, not truly, but
+// then we would have to replicate the transform at the place where
+// the mapping is used. I prefer to do it only once.
+type RefMap map[string]Ref
+
+// MakeSecrets creates an array of new Secrets filled with the
+// key/value pairs from the specified map. It further returns a map
+// showing which original CV name maps to what secret+key combination.
+func MakeSecrets(secrets model.CVMap, defaults map[string]string) ([]*apiv1.Secret, RefMap, error) {
+	var thesecrets []*apiv1.Secret
+	refs := make(map[string]Ref)
+
+	max := apiv1.MaxSecretSize
+	count := 1
+	current := NewSecret(fmt.Sprintf("secret-%d", count))
```

Comment:
This is not really a file, even it is later dumped into one.
`currentSecret` ? `currentK8sSecret` ?